core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -2246,30 +2251,31 @@ object UnifiedLog extends Logging {
                                           config: LogConfig,
                                           scheduler: Scheduler,
                                           logDirFailureChannel: LogDirFailureChannel,
-                                          logPrefix: String): SplitSegmentResult = {
-    LocalLog.splitOverflowedSegment(segment, existingSegments, dir, topicPartition, config, scheduler, logDirFailureChannel, logPrefix)
+                                          logger: Logger): SplitSegmentResult = {
+    LocalLog.splitOverflowedSegment(segment, existingSegments, dir, topicPartition, config, scheduler, logDirFailureChannel, logger)
   }
 
-  private[log] def deleteProducerSnapshots(segments: Iterable[LogSegment],
+  private[log] def deleteProducerSnapshots(segments: util.Collection[LogSegment],
                                            producerStateManager: ProducerStateManager,
                                            asyncDelete: Boolean,
                                            scheduler: Scheduler,
                                            config: LogConfig,
                                            logDirFailureChannel: LogDirFailureChannel,
                                            parentDir: String,
                                            topicPartition: TopicPartition): Unit = {
-    val snapshotsToDelete = segments.flatMap { segment =>
-      producerStateManager.removeAndMarkSnapshotForDeletion(segment.baseOffset).asScala
-    }
+    val snapshotsToDelete = segments.stream().flatMap { segment =>
+      val snapshotOptional = producerStateManager.removeAndMarkSnapshotForDeletion(segment.baseOffset)
+      if (snapshotOptional.isPresent) util.stream.Stream.of[SnapshotFile](snapshotOptional.get) else util.stream.Stream.empty[SnapshotFile]
```

Comment:
Nit: does `snapshotOptional.map(Stream::of).orElseGet(Stream::empty);` work? Also, it can probably be inlined (i.e. no need for the variable `snapshotOptional`.